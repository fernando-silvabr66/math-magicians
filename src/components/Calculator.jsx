import React, { Component } from 'react';
import {
  Container,
  Row,
  Col,
  Button,
} from 'react-bootstrap';
import calc from '../calc-events/calc';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      results: null,
      next: null,
      operation: null,
      valueToShow: 0,
    };
    this.clickHandler = this.clickHandler.bind(this);
  }

  clickHandler(e) {
    const newValues = calc(this.state, e.target.textContent);
    this.setState(() => newValues);
    const { results, next, operation } = newValues;
    if (next !== null && next !== undefined) {
      if (operation !== null && operation !== undefined) {
        const formula = `${results} ${operation} ${next}`;
        this.setState({ valueToShow: formula });
      } else this.setState({ valueToShow: next });
    } else if (next === null && operation !== null) this.setState({ valueToShow: operation });
    else this.setState({ valueToShow: (results !== null && results !== undefined) ? results : '0' });
  }

  render() {
    const { valueToShow } = this.state;
    return (
      <div className="calculator">
        <Container>
          <Row className="m-2">
            <Col>
              <div className="w-100 p-2 result">
                { valueToShow }
              </div>
            </Col>
          </Row>
          <Row className="m-2">
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>AC</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>+/-</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>%</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4 btn-operations" onClick={this.clickHandler}>รท</Button>
            </Col>
          </Row>
          <Row className="m-2">
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>7</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>8</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>9</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4 btn-operations" onClick={this.clickHandler}>
                x
              </Button>
            </Col>
          </Row>
          <Row className="m-2">
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>4</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>5</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>6</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4 btn-operations" onClick={this.clickHandler}>
                -
              </Button>
            </Col>
          </Row>
          <Row className="m-2">
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>1</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>2</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>3</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4 btn-operations" onClick={this.clickHandler}>
                +
              </Button>
            </Col>
          </Row>
          <Row className="m-2">
            <Col xs={6}>
              <Button className="w-100 p-4" onClick={this.clickHandler}>0</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4" onClick={this.clickHandler}>.</Button>
            </Col>
            <Col>
              <Button className="w-100 p-4 btn-operations" onClick={this.clickHandler}>=</Button>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

export default Calculator;
