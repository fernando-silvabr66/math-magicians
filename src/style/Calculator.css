.calculator {
  width: 60%;
}

.m-2 {
  margin: 0 !important;
}

.row > * {
  padding-right: 0 !important;
  padding-left: 0 !important;
}

.btn {
  border-radius: 0 !important;
  color: black;
}

.result {
  background-color: gray;
  color: white;
  line-height: 43px;
  min-height: 7vh;
  text-align: right;  
}

.p-4 {
  background-color: #e0e0e0;
  border-color: #dbdbdb;
}

.btn-operations {
  background-color: #f5913e;
  border: 1px solid #dbdbdb;
  color: black;
}

@media screen and (max-width: 800px) {
  .calculator-section {
    flex-direction: column;
  }

  .calculator {
    width: 100%;
  }
}
