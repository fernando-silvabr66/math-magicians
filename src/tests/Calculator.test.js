import renderer from 'react-test-renderer';
import Calculator from '../pages/CalculatorPage';

it('check calculator render correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
