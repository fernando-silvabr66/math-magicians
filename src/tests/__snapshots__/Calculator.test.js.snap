// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check calculator render correctly 1`] = `
<section
  className="container calculator-section"
>
  <h2>
     Let's do some math ! 
  </h2>
  <div
    className="calculator"
  >
    <div
      className="calculator"
    >
      <div
        className="container"
      >
        <div
          className="m-2 row"
        >
          <div
            className="col"
          >
            <div
              className="w-100 p-2 result"
            >
              0
            </div>
          </div>
        </div>
        <div
          className="m-2 row"
        >
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn-operations btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </div>
        </div>
        <div
          className="m-2 row"
        >
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn-operations btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              x
            </button>
          </div>
        </div>
        <div
          className="m-2 row"
        >
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn-operations btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </div>
        </div>
        <div
          className="m-2 row"
        >
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn-operations btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </div>
        </div>
        <div
          className="m-2 row"
        >
          <div
            className="col-6"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </div>
          <div
            className="col"
          >
            <button
              className="w-100 p-4 btn-operations btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
